test/postControllerTest.js

```diff
@@ -0,0 +1,96 @@
+/* global it, describe */
+const express = require('express');
+const request = require('supertest');
+const postController = require('../controllers/postController');
+const sinon = require('sinon');
+const postService = require('../services/postService');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(express.static('../public'));
+
+app.use('', postController);
+const server = app.listen(3000);
+
+
+describe('postController', () => {
+  describe('GET /', () => {
+    it('should respond with homepage', (done) => {
+      const stub = sinon.stub(postService, 'getAllPosts').resolves('expected');
+      request(app)
+        .get('/')
+        .expect(200)
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(/My First Blog/)
```

Comment:
This assert doesn't test post controller, in fact it is testing nav

Reply:
Then in order to test postController, should I be testing if the postService method is called correctly? Also, should nav testing be in a separate test? Since my postController doesn't return a value, testing nav is the equivalent of testing if my postController is returning correctly, yes?

Reply:
Since you are mocking out `postService.getAllPosts`, then you should definitely test if the mock is called correctly.

The main thing that your controller does when handing `GET /` is passing `posts` to a view. That is the thing that you should test.

Testing that nav is rendered for `GET /` is probably more fit for an e2e test